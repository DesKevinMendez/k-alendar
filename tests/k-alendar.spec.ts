import kAlendar from '@/components/KAlendar.vue';
import { mount, VueWrapper } from '@vue/test-utils';
import MockDate from 'mockdate';
import { beforeEach, describe, expect, it } from 'vitest';
const fakeDate = '2024-08-22T03:59:10.000000Z';

const props = {
  events: [
    {
      id: '9cc69d03-7215-41e2-966a-9c787b690223',
      title: 'Dto',
      description: 'Das',
      color: '#a02222',
      start_date: '2024-08-16T00:35:00.000000Z',
      end_date: '2024-08-17T00:35:00.000000Z'
    },
  ]
}
let wrapper: VueWrapper<InstanceType<typeof kAlendar>>

describe('kAlendar', () => {

  beforeEach(() => {
    MockDate.set(fakeDate)
    wrapper = mount(kAlendar, {
      props,
    });
  })

  beforeEach(() => {
    MockDate.reset();
  });

  it('should render the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
