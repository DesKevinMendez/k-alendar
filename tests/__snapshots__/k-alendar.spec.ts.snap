// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`kAlendar > should render the component 1`] = `
"<section data-v-74bbb8b8="" class="k-alendar-wrapper-container">
  <header data-v-0a815d1e="" data-v-74bbb8b8="" class="k-alendar-header-container">
    <div data-v-0a815d1e="" class="left-buttons">
      <div data-v-0a815d1e=""><button data-v-11092de0="" data-v-0a815d1e="" type="button" class="k-alendar-navegation-prev">←</button><button data-v-f3d09f65="" data-v-0a815d1e="" type="button" class="k-alendar-navegation-left">→</button></div><button data-v-8f6b2986="" data-v-0a815d1e="" type="button" class="k-alendar-today-button">Today</button>
    </div>
    <div data-v-0a815d1e="" class="center-title">
      <h2 data-v-0a815d1e="">October 2024</h2>
    </div>
  </header>
  <div data-v-a8a4dcb5="" data-v-74bbb8b8="" class="k-alendar-days-container">
    <div data-v-a8a4dcb5="">Mon</div>
    <div data-v-a8a4dcb5="">Tue</div>
    <div data-v-a8a4dcb5="">Wed</div>
    <div data-v-a8a4dcb5="">Thu</div>
    <div data-v-a8a4dcb5="">Fri</div>
    <div data-v-a8a4dcb5="">Sat</div>
    <div data-v-a8a4dcb5="">Sun</div>
  </div>
  <div data-v-6548dee6="" data-v-74bbb8b8="" class="k-alendar-container">
    <div data-v-6548dee6="" class="other-month-date date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">30</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">1</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">2</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">3</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">4</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">5</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">6</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">7</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">8</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">9</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">10</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">11</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">12</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">13</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">14</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">15</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">16</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">17</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">18</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">19</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">20</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">21</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">22</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">23</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">24</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">25</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">26</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">27</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">28</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">29</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">30</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">31</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="other-month-date date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">1</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="other-month-date date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">2</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
    <div data-v-6548dee6="" class="other-month-date date">
      <div data-v-6548dee6="" class="k-alendar-span-container"><span data-v-6548dee6="" class="k-alendar-text">3</span>
        <!--v-if-->
      </div>
      <!--v-if-->
    </div>
  </div>
  <dialog data-v-a389628f="" data-v-74bbb8b8="" id="k-alendar-dialog" modal-mode="mega" class="k-alendar-dialog" style="top: 0px; left: 0px;">
    <header>
      <div data-v-a389628f="" class="flex justify-between">
        <!--v-if-->
        <!--v-if-->
      </div><button class="k-alendar-button-close" type="button"><svg aria-hidden="true" focusable="false" data-prefix="fas" data-icon="xmark" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 320 512">
          <path class="" fill="currentColor" d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"></path>
        </svg></button>
    </header>
    <article class="k-alendar-dialog-main">
      <section data-v-a389628f="" class="k-alendar-event-detail-main-wrapper">
        <!--v-if-->
        <div data-v-a389628f="" class="dates"><time data-v-a389628f="" datetime=""></time></div>
        <!--v-if-->
        <!--v-if-->
      </section>
    </article>
    <footer></footer>
  </dialog>
</section>"
`;
